website/docs/guides/migrating-from-v2-to-v3.mdx

```diff
@@ -0,0 +1,87 @@
+# Migrating from `v2` to `v3`
+
+With new major version comes breaking changes. This section will guide you through the process of migrating from `v2` to `v3`.
+
+### 1. Remove `graphql` as a peer dependency
+
+We have designed the new `envelop` to be engine agnostic. This allows you to use any GraphQL engine you want. This means that `graphql` is no longer a peer dependency and `envelop` simply just wraps the `parse`, `validate`, `execute` and `subscribe` functions that you provide.
+
+```diff
+import { envelop } from '@envelop/core';
++ import { parse, validate, execute, subscribe } from 'graphql';
+
+- const getEnveloped = envelop([ ... ])
++ const getEnveloped = envelop({ parse, validate, execute, subscribe, plugins: [ ... ] })
+```
+
+### 2. Remove `useAsyncSchema` plugin
+
+This was a mistake from beginning as we cannot asynchronously `validate` and `parse` since with [`graphql`](https://github.com/graphql/graphql-js) these functions are synchronous in nature.
+
+### 3. Rename `useLazyLoadedSchema` to `useSchemaByContext`
+
+Oringinal name was very misleading since lazy loading could mean it can be asynchronous in nature. This plugin was renamed to better reflect its purpose. It is now called `useSchemaByContext`
+
+### 4. Remove `enableIf` utility
+
+This utility was used to enable plugins conditionally. For a better developer experience we have dropped this utility and favor more type safe way to conditionally enable plugins.
+
+```diff
+- import { envelop, useMaskedErrors, enableIf } from '@envelop/core'
++ import { envelop, useMaskedErrors } from '@envelop/core'
+import { parse, validate, execute, subscribe } from 'graphql'
+
+const isProd = process.env.NODE_ENV === 'production'
+
+const getEnveloped = envelop({
+	parse,
+   validate,
+   execute,
+   subscribe,
+   plugins: [
+     // This plugin is enabled only in production
+-    enableIf(isProd, useMaskedErrors())
++    isProd && useMaskedErrors()
+   ]
+ })
+```
+
+### 5. Update options for `useMaskedErrors` plugin
+
+- We _removed_ `handleValidationErrors` and `handleParseErrors` options since ONLY masking validation errors OR ONLY disabling introspection errors does not make sense, as both can be abused for reverse-engineering the GraphQL schema (see https://github.com/nikitastupin/clairvoyance for reverse-engineering the schema based on validation error suggestions). Instead you should use `useErrorHandler` plugin where you can access each phase and decide what to do with the error.
+- Renamed `formatError` to `maskError`
+
+### 6. Drop `useTiming` plugin
```

Comment:
What do we tell people that still want to use it?

Reply:
We don’t have alternative for this yet and honestly don’t think we should make this for real world prod usage most people would use sentry, Prometheus or Hive to get the resolver usage timings